sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -563,7 +573,17 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   private[this] def castToFloat(from: DataType): Any => Any = from match {
     case StringType =>
       buildCast[UTF8String](_, s => try s.toString.toFloat catch {
-        case _: NumberFormatException => null
+        case _: NumberFormatException =>
+          val str = s.trim.toString
+          if (str.equalsIgnoreCase("infinity")) {
```

Comment:
just a check; we don't support some alternatives, e.g., inf, -inf, ... ?https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/float.c#L194

Reply:
@maropu We don't support inf and -inf, should we support ?

Reply:
Is there another db-like systems supporting this, e.g., DB2?

Reply:
@maropu Yes.. db2 does allow inf and -inf. Lets add in spark also ?

Reply:
IMO some users might think `inf` is more common, e.g., even in python, both `inf` and `infinity` can be used;
```
Python 3.6.8 |Anaconda, Inc.| (default, Dec 29 2018, 19:04:46) 
>>> float('inf')
inf
>>> float('infinity')
inf
```
Its ok to wait for other reviewer's comments. @gatorsmile @dongjoon-hyun

Reply:
@gatorsmile @dongjoon-hyun Could you please help make a decision to support `inf`, `+inf` for positive infinity and '-inf` for negative infinity ? Thanks a lot. Thanks again @maropu for pointing this.

Reply:
As @maropu mentioned, PostgreSQL supports that, I'm +1 for supporting it.
```
		else if (pg_strncasecmp(num, "inf", 3) == 0)
		{
			val = get_float4_infinity();
			endptr = num + 3;
		}
		else if (pg_strncasecmp(num, "+inf", 4) == 0)
		{
			val = get_float4_infinity();
			endptr = num + 4;
		}
		else if (pg_strncasecmp(num, "-inf", 4) == 0)
		{
			val = -get_float4_infinity();
			endptr = num + 4;
		}
```

Reply:
@dongjoon-hyun OK.. thanks.. i will make the change.